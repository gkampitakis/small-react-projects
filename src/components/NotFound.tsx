import React, { ReactElement } from 'react';
import { Link } from 'react-router-dom';
import NotFoundImg from './images/404.svg';


export default function NotFound (): ReactElement {
  return (
    <main className="not_found">
      <img src={NotFoundImg} alt="Not found image" />
      <p>The page you requested was not found</p>
      <div>
        <Link to="/" className="return-btn">return home</Link>
      </div>
    </main>
  );
}
