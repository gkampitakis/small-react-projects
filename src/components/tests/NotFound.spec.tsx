import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import NotFound from '../NotFound';


describe('NotFound', () => {
  it('Should render correctly', () => {
    const { baseElement } = render(<NotFound />, { wrapper: MemoryRouter });

    expect(baseElement).toMatchSnapshot();
  });
});
