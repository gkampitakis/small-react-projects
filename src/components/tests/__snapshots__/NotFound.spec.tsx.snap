// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Should render correctly 1`] = `
<body>
  <div>
    <main
      class="not_found"
    >
      <img
        alt="Not found image"
        src="./images/404.svg"
      />
      <p>
        The page you requested was not found
      </p>
      <div>
        <a
          class="return-btn"
          href="/home"
        >
          return home
        </a>
      </div>
    </main>
  </div>
</body>
`;
